### PR TITLE
EZP-29833 : Limiting "Content\Create" to Section disables "Publish" button

### PR DESCRIPTION
**Pull request by pawbuj**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-29833](https://jira.ez.no/browse/EZP-29833)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Fixes permission check for content creation when rights are limited some section.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by ViniTou** [comment]
Also, not sure how all of you are feeling about creating those structs inside menuBuilder instead of passing them as options.
@webhdx 
@Nattfarinn 
@adamwojs

**Comment by pawbuj** [comment]
The main reason to add createContentCreateStruct function was to increase readability of the code.

File: src/lib/Menu/ContentCreateRightSidebarBuilder.php [hunk 1]
```diff
@@ -171,4 +175,19 @@ public static function getTranslationMessages(): array
...
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     * @param \eZ\Publish\API\Repository\Values\Content\Language $language
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\ContentCreateStruct
+     */
+    private function createContentCreateStruct(Location $location, ContentType $contentType, Language $language)
```
[the remark this answers is not in the document]
Comment (pawbuj) [reply src/lib/Menu/ContentCreateRightSidebarBuilder.php]: done

**Comment by micszo** [comment]
@pawbuj accoring to _Affects Version/s_ base needs to changed to 1.3. Also please rebase afterwards.

**Comment by pawbuj** [comment]
> @pawbuj according to _Affects Version/s_ base needs to be changed to 1.3. Also please rebase afterwards.

donw

**Comment by pawbuj** [comment]
@micszo yes, sure we should also fix this, but I think that we should have a new task for this in jira.

**Comment by lserwatka** [comment]
@pawbuj we don’t create tickets per affected version. You simply set right target branch (1.3 in your case) and after merge we merge it up to master.

**Comment by pawbuj** [comment]
I mean a new ticket for the button in the modal (screenshot by @micszo ). Target branch is already changed to 1.3.

**Comment by micszo** [comment]
@pawbuj there you go https://jira.ez.no/browse/EZP-30001.

**Comment by lserwatka** [comment]
You can merge it up

**Pull request merged**